cpp/src/plasma/protocol.cc

```diff
@@ -230,35 +230,63 @@ Status ReadReleaseReply(uint8_t* data, size_t size, ObjectID* object_id) {
   return plasma_error_status(message->error());
 }
 
-// Delete messages.
+// Delete objects messages.
 
-Status SendDeleteRequest(int sock, ObjectID object_id) {
+Status SendDeleteRequest(int sock, const std::vector<ObjectID>& object_ids) {
   flatbuffers::FlatBufferBuilder fbb;
-  auto message = CreatePlasmaDeleteRequest(fbb, fbb.CreateString(object_id.binary()));
+  auto message =
+      CreatePlasmaDeleteRequest(fbb, static_cast<int>(object_ids.size()),
+                                to_flatbuffer(&fbb, &object_ids[0], object_ids.size()));
   return PlasmaSend(sock, MessageType::PlasmaDeleteRequest, &fbb, message);
 }
 
-Status ReadDeleteRequest(uint8_t* data, size_t size, ObjectID* object_id) {
+Status ReadDeleteRequest(uint8_t* data, size_t size, std::vector<ObjectID>& object_ids) {
   DCHECK(data);
-  auto message = flatbuffers::GetRoot<PlasmaReleaseReply>(data);
+  auto message = flatbuffers::GetRoot<PlasmaDeleteRequest>(data);
   DCHECK(verify_flatbuffer(message, data, size));
-  *object_id = ObjectID::from_binary(message->object_id()->str());
+  int num_objects = message->count();
+  object_ids.clear();
```

Comment:
This pattern here is repeated a number of times in this PR. Let's introduce a from_flatbuffers equivalently to https://github.com/apache/arrow/blob/master/cpp/src/plasma/protocol.cc#L35 (but using std::vector instead of arrays) to do this. Also remember https://google.github.io/styleguide/cppguide.html#Reference_Arguments for this.